mrmustard/math/backend_numpy.py

```diff
@@ -84,14 +84,14 @@ def arange(
     def asnumpy(self, tensor: np.ndarray) -> np.ndarray:
         if isinstance(tensor, np.ndarray):
             return tensor
-        return np.array(tensor)
+        return np.asarray(tensor)
 
     def assign(self, tensor: np.ndarray, value: np.ndarray) -> np.ndarray:
         tensor = value
         return tensor
 
     def astensor(self, array: np.ndarray, dtype=None) -> np.ndarray:
-        array = np.array(array)
+        array = np.asarray(array)
         return self.cast(array, dtype=dtype or array.dtype)
```

Comment:
I think we can simplify this to 
```suggestion
        return np.asarray(array, dtype=dtype)
```